packages/alfa-rules/src/common/predicate/is-clipped.ts

```diff
@@ -0,0 +1,74 @@
+import { Cache } from "@siteimprove/alfa-cache";
+import { Device } from "@siteimprove/alfa-device";
+import { Element, Text, Node } from "@siteimprove/alfa-dom";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Context } from "@siteimprove/alfa-selector";
+import { Style } from "@siteimprove/alfa-style";
+
+const { abs } = Math;
+const { isElement } = Element;
+const { isText } = Text;
+
+const cache = Cache.empty<Device, Cache<Context, Cache<Node, boolean>>>();
+
+export function isClipped(
+  device: Device,
+  context: Context = Context.empty()
+): Predicate<Node> {
+  return function isClipped(node): boolean {
+    return cache
+      .get(device, Cache.empty)
+      .get(context, Cache.empty)
+      .get(node, () => {
+        if (isElement(node)) {
+          const style = Style.from(node, device, context);
+
+          const { value: height } = style.computed("height");
+          const { value: width } = style.computed("width");
+          const { value: x } = style.computed("overflow-x");
+          const { value: y } = style.computed("overflow-y");
+
+          if (
+            height.type === "length" &&
```

Comment:
Also, wondering whether we should check for `height: 0%`. But maybe that's not really used for clipping.

Reply:
Good point!